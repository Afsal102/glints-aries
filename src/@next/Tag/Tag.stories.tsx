import { Meta, Story } from '@storybook/react';
import React from 'react';

import { BaseContainer } from '../../Layout/GlintsContainer/GlintsContainer';
import { Tag, TagProps } from './Tag';

(Tag as React.FunctionComponent<TagProps>).displayName = 'Tag';

const defaultText = 'Basic Tag';

export default {
  title: '@next/Tag',
  component: Tag,
  decorators: [Story => <BaseContainer>{Story()}</BaseContainer>],
} as Meta;

const Template: Story<TagProps> = args => <Tag {...args}>{defaultText}</Tag>;

export const Interactive = Template.bind({});

Interactive.args = {};
