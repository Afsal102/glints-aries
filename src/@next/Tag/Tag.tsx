import React from 'react';
import { Typography } from '../Typography';
import { Neutral } from '../utilities/colors';
import { TagStyle } from './TagStyle';

export interface TagProps extends React.HTMLAttributes<HTMLDivElement> {
  children?: React.ReactNode;
}

export const Tag = ({ children, ...props }: TagProps) => {
  const content =
    typeof children === 'string' ? (
      <Typography variant="caption" color={Neutral.B18} as={'span'}>
        {children}
      </Typography>
    ) : (
      children
    );

  return <TagStyle {...props}>{content}</TagStyle>;
};
