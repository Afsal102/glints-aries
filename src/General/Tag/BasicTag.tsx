import * as React from 'react';
import classNames from 'classnames';
import { TagContainer, TagContent } from './TagStyle';

const Tag: React.FunctionComponent<Props> = props => {
  const { className, children, block, outline, onClick, ...restProps } = props;

  return (
    <TagContainer
      className={classNames('aries-tag', className)}
      role="presentation"
      tabIndex={0}
      block={block}
      outline={outline}
      isClickable={typeof onClick === 'function'}
      onClick={onClick}
      {...restProps}
    >
      <TagContent className="tag-content" tabIndex={-1}>
        {children}
      </TagContent>
    </TagContainer>
  );
};

Tag.defaultProps = {
  block: false,
  outline: false,
};

export type Props = React.ComponentPropsWithoutRef<typeof TagContainer> & {
  children: React.ReactNode;
};

export default Tag;
