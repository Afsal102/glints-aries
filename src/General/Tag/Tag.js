/* @flow */

import React, { Fragment } from 'react';
import { TagContainer } from '../../Style/General/TagStyle';

const Tag = (props: Props) => {
  const {
    label,
    className,
    children,
    ...defaultProps
  } = props;

  return (
    <Fragment>
      <TagContainer className={className} {...defaultProps}>
        {label}
        {children}
      </TagContainer>
    </Fragment>
  );
};

type Props = {
  children: React$Node,
  label: string,
  className: string,
}

export default Tag;
