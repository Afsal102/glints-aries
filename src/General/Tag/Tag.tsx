import * as React from 'react';
import classNames from 'classnames';
import {
  TagContainer,
  TagContent,
  StartIconContainer,
  EndIconContainer,
} from '../../Style/General/TagStyle';

const Tag: React.FunctionComponent<Props> = props => {
  const {
    className,
    children,
    theme,
    block,
    outline,
    borderStyle,
    startIcon,
    endIcon,
    onClick,
    ...restProps
  } = props;

  const content = (
    <>
      {startIcon && <StartIconContainer>{startIcon}</StartIconContainer>}
      {children}
      {endIcon && <EndIconContainer>{endIcon}</EndIconContainer>}
    </>
  );

  return (
    <React.Fragment>
      <TagContainer
        className={classNames('aries-tag', className)}
        role="presentation"
        tabIndex={0}
        theme={theme}
        block={block}
        outline={outline}
        borderStyle={borderStyle}
        {...restProps}
      >
        <TagContent
          className="tag-content"
          tabIndex={-1}
          isClickable={typeof onClick === 'function'}
          onClick={onClick}
        >
          {content}
        </TagContent>
      </TagContainer>
    </React.Fragment>
  );
};

Tag.defaultProps = {
  theme: 'grey',
  block: false,
  outline: false,
  borderStyle: 'solid',
};

interface Props extends React.ComponentPropsWithoutRef<typeof TagContainer> {
  children: React.ReactNode;
  theme?: string;
  onClick?(e: React.MouseEvent<HTMLLabelElement, MouseEvent>): void;
  startIcon?: React.ReactNode;
  endIcon?: React.ReactNode;
}

export default Tag;
