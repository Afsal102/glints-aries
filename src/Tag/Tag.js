/* @flow */

import React from 'react';
import { Container } from '../Style/TagStyle';

const Tag = (props: Props) => {
  const {
    label,
    className,
    children,
    ...defaultProps
  } = props;

  return (
    <Container className={className} {...defaultProps}>
      {label}
      {children}
    </Container>
  );
};

type Props = {
  children: React$Node,
  label: string,
  className: string,
}

export default Tag;
