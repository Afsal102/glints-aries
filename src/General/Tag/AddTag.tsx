import styled from 'styled-components';

import { Greyscale, SecondaryColor } from '../../Style/Colors';
import BasicTag from './BasicTag';

export const AddTag = styled(BasicTag)`
  border-style: dashed;
  border-color: ${SecondaryColor.lightgrey};

  &:hover {
    border-color: ${Greyscale.grey};
  

  &:active {
    border-color: ${Greyscale.black};
  }
`;
