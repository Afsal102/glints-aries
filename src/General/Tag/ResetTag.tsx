import styled from 'styled-components';

import { PrimaryColor } from '../../Style/Colors';
import BasicTag from './BasicTag';

export const ResetTag = styled(BasicTag)`
  color: ${PrimaryColor.glintsred};
  border-style: solid;
  border-color: transparent;
  background-color: rgba(236, 39, 43, 0.05);

  &:hover {
    background-color: rgba(236, 39, 43, 0.1);
  }

  &:active {
    background-color: rgba(236, 39, 43, 0.2);
  }
`;
