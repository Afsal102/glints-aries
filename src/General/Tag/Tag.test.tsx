import * as React from 'react';

import * as renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

import Tag from './Tag';

const children = 'Glints Aries';

it(`<Tag> should render a label with the text "Glints Aries"`, () => {
  const TagSnapshot = renderer.create(<Tag>{children}</Tag>).toJSON();
  expect(TagSnapshot).toMatchSnapshot();
});

it('should display the correct text when rendered', () => {
  const { getByText } = render(<Tag>{children}</Tag>);
  const tagContent = getByText(children);
  expect(tagContent).toBeTruthy();
});

describe('when block is:', () => {
  it('true, it should not have a border-radius property', () => {
    const { getByRole } = render(<Tag block>{children}</Tag>);
    const tagContainer = getByRole('presentation');
    const { borderRadius } = getComputedStyle(tagContainer);
    expect(borderRadius).toBeFalsy();
  });

  it('false, it should have a border-radius property', () => {
    const { getByRole } = render(<Tag>{children}</Tag>);
    const tagContainer = getByRole('presentation');
    const { borderRadius } = getComputedStyle(tagContainer);
    expect(borderRadius).toBeTruthy();
  });
});

describe('when outline is:', () => {
  it('true, should not have a background property', () => {
    const { getByRole } = render(<Tag outline>{children}</Tag>);
    const tagContainer = getByRole('presentation');
    const { background } = getComputedStyle(tagContainer);
    expect(background).toBeFalsy();
  });

  it('false, should have a background property', () => {
    const { getByRole } = render(<Tag>{children}</Tag>);
    const tagContainer = getByRole('presentation');
    const { background } = getComputedStyle(tagContainer);
    expect(background).toBeTruthy();
  });
});
