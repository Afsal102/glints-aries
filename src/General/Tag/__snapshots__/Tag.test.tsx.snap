// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag> should render a label with the text "Glints Aries" 1`] = `
<div
  className="TagStyle__TagContainer-sc-1idy1a0-1 dejuZz aries-tag"
  role="presentation"
  tabIndex={0}
>
  <label
    className="TagStyle__TagContent-sc-1idy1a0-0 DLdwx tag-content"
    tabIndex={-1}
  >
    Glints Aries
  </label>
</div>
`;
